fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/simplification/MatchSimpleFieldValueRule.java

```diff
@@ -68,6 +69,9 @@ public void onMatch(@Nonnull final ValueComputationRuleCall<Value, Map<Value, Va
         if (!rootValue.isFunctionallyDependentOn(baseValue)) {
             return;
         }
+        if (rootValue.getResultType().equals(PseudoField.ROW_VERSION.getType())) {
```

Comment:
Wouldn't that prevent us from being able to copy that over in the covering-to-partial codepath?